### PR TITLE
fix(connectors): fix https://issues.redhat.com/browse/MGDCTRS-834

### PR DESCRIPTION
**Pull request by valdar**
## Description
The problem was happening specifically in case of expired namespaces that had connectors previously assigned.

The namespace expires and its status is set to deleting, the control plane delete the namespace and the status is updated to deleted. At this point  the namespace reconcile loop should run and correctly delete the namespace, unfortunately what happens inside the reconcile loop is that the namespace is re-detected as expired and placed back in deleting state before actually trying to delete it. In a never ending loop.

This PR fix this behavior.

See https://issues.redhat.com/browse/MGDCTRS-834

## Verification Steps

1. OCM_ENV=integration make test/integration/connector     

## Checklist (Definition of Done)
- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [x] ~~Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [x] Code Review completed
- [x] Verified independently by reviewer
- [x] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [x] ~~Required Standard Operating Procedure (SOP) is added.~~
- [x] ~~JIRA has created for changes required on the client side~~

### CODE REVIEW

**Comment by dhirajsb** [comment]
Can you resolve conflicts?

File: internal/connector/internal/services/connector_namespaces.go [hunk 1]
```diff
@@ -484,9 +485,8 @@ func (k *connectorNamespaceService) ReconcileDeletedNamespaces() (int, []*errors
 		if err := dbConn.Table("connector_namespaces").Select("connector_namespaces.id").
 			Joins("LEFT JOIN connector_statuses ON connector_statuses.namespace_id = connector_namespaces.id AND "+
 				"connector_statuses.deleted_at IS NULL").
-			Group("connector_namespaces.id").
```
Comment (dhirajsb) [inline internal/connector/internal/services/connector_namespaces.go:487]: We also need this fix in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/bd75b56ea197b53e97fe1abbacad029c9f1ca361/internal/connector/internal/services/connector_cluster.go#L860

**Review by dhirajsb — CHANGES_REQUESTED** [review]

File: internal/connector/internal/services/connector_namespaces.go [hunk 2]
```diff
@@ -312,7 +313,7 @@ func (k *connectorNamespaceService) GetExpiredNamespaceIds() ([]string, *errors.
 	dbConn := k.connectionFactory.New()
 	now := time.Now()
 	if err := dbConn.Model(&dbapi.ConnectorNamespace{}).Select("id").
-		Where("expiration < ?", now).Find(&result).Error; err != nil {
+		Where("expiration < ? AND status_phase NOT IN ?", now, []string{string(dbapi.ConnectorNamespacePhaseDeleting), string(dbapi.ConnectorNamespacePhaseDeleted)}).Find(&result).Error; err != nil {
```
Comment (dhirajsb) [inline internal/connector/internal/services/connector_namespaces.go:316]: We also need to exclude `deleting` and `deleted` namespaces in another update https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4a31a0543d271107e97fd74482bd870afefc2dac/internal/connector/internal/services/connector_namespaces.go#L389

File: internal/connector/test/integration/features/connector-multitenancy-api.feature [hunk 3]
```diff
@@ -201,12 +201,71 @@ Feature: connector namespaces API
...
 
-    # Eval namespace should expire and get deleted after 2 seconds as configured in internal/connector/test/integration/feature_test.go:27
-    Given I wait up to "10" seconds for a GET on path "/v1/kafka_connector_namespaces/" response ".total" selection to match "0"
+     # Eval namespace should expire and get deleted after 15 seconds as configured in internal/connector/test/integration/feature_test.go:27
+    Given I wait up to "20" seconds for a GET on path "/v1/kafka_connector_namespaces/" response ".total" selection to match "0"
     And I GET path "/v1/kafka_connector_namespaces/"
     Then the response code should be 200
     And the ".total" selection from the response should match "0"
 
+
```
[the remark this answers is not in the document]
Comment (valdar) [reply internal/connector/test/integration/features/connector-multitenancy-api.feature:210]: So to be precise: this approach, although more convoluted, exposes another new bug that your other suggestion would fix...

File: internal/connector/internal/services/connector_namespaces.go (hunk 2 above)
Comment (valdar) [reply internal/connector/internal/services/connector_namespaces.go:318]: This should fix the not yet seen bug that the test approach you proposed should expose. Correct?

Comment (dhirajsb) [reply internal/connector/internal/services/connector_namespaces.go:318]: yes

File: internal/connector/test/integration/features/connector-multitenancy-api.feature (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhirajsb) [reply internal/connector/test/integration/features/connector-multitenancy-api.feature:210]: exactly, white box testing ftw!

[the remark this answers is not in the document]
Comment (dhirajsb) [reply internal/connector/test/integration/features/connector-multitenancy-api.feature]: Refactored the test to reproduce the error in `connector-agent-api.feature` and verified the fix resolves it.

**Review by dhirajsb — APPROVED** [review]
lgtm

**Pull request merged**